invertedai/common.py

```diff
@@ -141,22 +141,35 @@ class AgentAttributes(BaseModel):
     AgentState
     """
 
-    length: float  #: Longitudinal extent of the agent, in meters.
-    width: float  #: Lateral extent of the agent, in meters.
+    length: Optional[float] = None  #: Longitudinal extent of the agent, in meters.
+    width: Optional[float] = None  #: Lateral extent of the agent, in meters.
     #: Distance from the agent's center to its rear axis in meters. Determines motion constraints.
-    rear_axis_offset: float
+    rear_axis_offset: Optional[float] = None
+    agent_type: Optional[str] = 'car'  #: Type of agent, default to car, can also be pedestrian
```

Comment:
Can you make the type here the Enum instead of str ?

Reply:
I'm actually not sure that an enum is preferable. It would force people to update the package whenever we introduce additional types.

Reply:
If we are using string enums it will be automatically casted as string and when we introduce new types we just append more values. Plus it is  using enums for validation already on client side

Reply:
keeping string as per our conversation.